src/Autofac/Features/ResolveAnything/AnyConcreteTypeNotAlreadyRegisteredSource.cs

```diff
@@ -91,16 +91,39 @@ public AnyConcreteTypeNotAlreadyRegisteredSource(Func<Type, bool> predicate)
                 return Enumerable.Empty<IComponentRegistration>();
             }
 
+            // Check to resolve issue #925
             if (typeInfo.IsGenericType)
             {
-                foreach (var typeParameter in typeInfo.GenericTypeArguments)
+                if (typeInfo.GetGenericTypeDefinition() == typeof(Lazy<>))
                 {
-                    // Issue #855: If the generic type argument doesn't match the filter don't look for a registration.
-                    if (_predicate(typeParameter) && !registrationAccessor(new TypedService(typeParameter)).Any())
```

Comment:
It looks like the predicate check here got removed in the new code. I recognize that if the tests are passing that we potentially don't have tests around that, though the comment here indicates the predicate was added for a reason.

Do we no longer need that? Are we missing a test?

Reply:
So I think the predicate was originally added to address [this issue](https://github.com/autofac/Autofac/issues/855). When I remove the predicate from the original code, `ConstructableOpenGenericsWithGenericTypeArgumentNotMatchingFilterCanBeResolved()` fails so it at least has one test. 

I don't think its needed anymore, but maybe I'm missing something. The test would originally fail because ACTNARS checked each type parameter for a registration when resolving a generic type. Since the `Exception` type did not satisfy the predicate, the whole `Progress<Exception>` type would fail to register without the predicate check. 

The new ACTNARS behavior passes the test because `Progress<Exception>` has a default constructor and it doesn't need a registration for `Exception`.